### PR TITLE
[RFR] React frontend authentication and orders

### PR DESCRIPTION
**Pull request by djhi**
- [x] Added API routes for signup
- [x] Added React frontend code for routes needing authentication
- [x] Added Orders routes

Anonymous access to order page redirect to login:
anonymous access to order

Authorized access to order:
authorized access to order

### CODE REVIEW

File: src/frontend/js/order/OrderItem.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    <Link to={`/orders/${id}`} className="list-group-item">
+        <span className={classNames('label label-default pull-xs-right', {
+            'label-info': status === 'pending',
+            'label-success': status === 'sent',
+            'label-danger': status === 'refused',
+        })}>{status}</span>
+        <span className="label label-default pull-xs-right">{numeral(total).format('$0.00')}</span>
+        <h4 className="list-group-item-heading">{reference} - <span className="text-muted">{moment(date).format('L')}</span></h4>
+        <p className="list-group-item-text">
+
```
Comment (Kmaschta) [inline src/frontend/js/order/OrderItem.js]: Empty item ?

File: src/frontend/js/order/orderApi.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/* globals API_URL */
+export function fetchOrders() {
+    return fetch(`${API_URL}/orders`, {
+        headers: {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json; charset=utf-8',
+            'Authorization': sessionStorage.getItem('token'), // TODO: this should probably be a function parameter
```
Comment (Kmaschta) [inline src/frontend/js/order/orderApi.js]: Probably.

File: src/frontend/js/order/orderReducer.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import { ORDERS_LOADED } from './orderActions';
+
+// [{
```
Comment (Kmaschta) [inline src/frontend/js/order/orderReducer.js]: What's the point of this comment ?

**Comment by Kmaschta** [comment]
Could you add screenshots when the PR will be switched to RFR ?

**Comment by djhi** [comment]
Sign-up UI will be addressed later

**Comment by alexisjanvier** [comment]
Unfortunately, my PR has been merged at first .... Need rebase !

**Comment by djhi** [comment]
rebased

**Comment by alexisjanvier** [comment]
:+1:

**Comment by Kmaschta** [comment]
:+1:

File: package.json [hunk 4]
```diff
@@ -103,6 +109,7 @@
     "redux-mock-store": "0.0.3",
     "selenium-standalone": "~4.8.0",
     "sinon": "2.0.0-pre",
+    "tether": "~1.1.1",
```
Comment (alexisjanvier) [inline package.json]: ?

Comment (djhi) [inline package.json]: needed by bootstrap

**Comment by djhi** [comment]
rebased

**Pull request merged**